GitUI/CommandsDialogs/FormCommit.cs

```diff
@@ -3309,6 +3344,61 @@ private void UpdateButtonStates()
                 : TranslatedStrings.ButtonPush;
         }
 
+        private void UICommands_PostRepositoryChanged(object sender, GitUIEventArgs e)
+        {
+            ThreadHelper.FileAndForget(async () =>
+                {
+                    await this.SwitchToMainThreadAsync();
+                    RescanChanges();
+                });
+        }
+
+        private bool ExecuteCommandStaged(int command)
+        {
+            IScriptsRunner scriptsRunner = UICommands.GetRequiredService<IScriptsRunner>();
+            return scriptsRunner.RunScript(command, owner: this, UICommands, new ScriptOptionsProvider(this, Staged));
+        }
+
+        private bool ExecuteCommandUnstaged(int command)
+        {
+            IScriptsRunner scriptsRunner = UICommands.GetRequiredService<IScriptsRunner>();
+            return scriptsRunner.RunScript(command, owner: this, UICommands, new ScriptOptionsProvider(this, Unstaged));
+        }
+
+        private class ScriptOptionsProvider : IScriptOptionsProvider
```

Comment:
Please move it into own file `FormCommit.ScriptOptionsProvider.cs`

Reply:
Accepted